editoast/src/valkey_utils.rs

```diff
@@ -202,19 +202,26 @@ pub enum ValkeyClient {
     NoCache,
 }
 
+pub struct ValkeyConfig {
+    /// Disables caching. This should not be used in production.
+    pub no_cache: bool,
+    pub is_cluster_client: bool,
+    pub valkey_url: Url,
+}
+
 impl ValkeyClient {
-    pub fn new(valkey_config: ValkeyConfig) -> Result<ValkeyClient> {
+    pub fn new(valkey_config: &ValkeyConfig) -> Result<ValkeyClient> {
         if valkey_config.no_cache {
             return Ok(ValkeyClient::NoCache);
         }
-        let valkey_config_url = valkey_config.url()?;
         if valkey_config.is_cluster_client {
             return Ok(ValkeyClient::Cluster(
-                redis::cluster::ClusterClient::new(vec![valkey_config_url.as_str()]).unwrap(),
+                redis::cluster::ClusterClient::new(vec![valkey_config.valkey_url.as_str()])
```

Comment:
If I understand well, `ClusterClient::new()` takes an iterator on anything that implements [`IntoConnectionInfo`](https://docs.rs/redis/latest/redis/trait.IntoConnectionInfo.html), which is the case for `Url` so I do not think you need the `.as_str()`, isn't it?

```suggestion
                redis::cluster::ClusterClient::new(vec![valkey_config.valkey_url])
```

Reply:
Done in [`3894923` (#9605)](https://github.com/OpenRailAssociation/osrd/pull/9605/commits/38949235d3746a6fae496d37ddfac81a98e89f34)

Reply:
Mmm, almost, there were 2 occurrences of this and you fixed the other one 😅

Reply:
My bad... done in [`c85db53` (#9605)](https://github.com/OpenRailAssociation/osrd/pull/9605/commits/c85db53ca27f2ce0465b8620690abef45fe29771) (hopefully this time xD)